### PR TITLE
SECURITY.md: Add initial security policy

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

RIOT should have a defined policy on how to handle security released bugs.

After a small survey of other OSes, this seems to fit best with RIOT.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
We need to come up with a trusted RIOT user link.

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# RIOT Security Policy
+
+All security bugs reported will be silently fixed in `master` and backported to the
+previous release.
+
+## Reporting a Vulnerability
+
+If a security issue is discovered, please report it to security@riot-os.org.
+A response will be provided within one week.
+The issue will be tracked in the mailing list.
```
Comment (waehlisch) [inline SECURITY.md:10]: be more specific. which mailing list?

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+The original reporter will be included in the issue.
```
Comment (waehlisch) [inline SECURITY.md:11]: you mean: "in the discussion of the issue"?

**Comment by MrKevinWeiss** [comment]
... The fact that this branch is on RIOT is somewhat intentional, any maintainer can push to if if they want.

**Review by waehlisch — APPROVED** [review]

File: SECURITY.md (hunk 1 above)
Comment (MrKevinWeiss) [reply SECURITY.md:10]: I thought security@riot-os.org if it is private.

File: SECURITY.md (hunk 2 above)
Comment (MrKevinWeiss) [reply SECURITY.md:11]: yup

File: SECURITY.md (hunk 1 above)
Comment (waehlisch) [reply SECURITY.md:10]: then: will be tracked via security@riot-os.org. ... or something like this.

**Comment by miri64** [comment]
@MrKevinWeiss Please squash.

**Comment by miri64** [comment]
BTW: If need be I can be added to the list of trusted RIOT users. The fingerprint for my public key is [7D11 80C6 578C 9EEA C85C B564 CF08 E639 7963 1D1C](https://keys.openpgp.org/vks/v1/by-fingerprint/7D1180C6578C9EEAC85CB564CF08E63979631D1C)

**Comment by MrKevinWeiss** [comment]
squashed

**Comment by MrKevinWeiss** [comment]
Added info on trusted riot users and squashed.  Read for final review.

File: SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+original reporter, RIOT security maintainers, and "Trusted RIOT Users".
+A public announcement of the security fix will be made two weeks after the
+point release, though this may vary depending on the severity and ability of
+trusted RIOT users to provide the fix.
+
+## Trusted RIOT Users
+
+To join the "Trusted RIOT Users" notifications list please send information
+on the RIOT based service or product to the
+[security mailing list](security@riot-os.org).
```
Comment (miri64) [inline SECURITY.md:25]: Why being so vague? IIRC we discussed offline that this "Trusted RIOT Users" notification list will be a dedicated category at https://forum.riot-os.org. IMHO we can (and should) be very open about this. As such, you should also mention, that we need their forum username with the information they send.

**Comment by MrKevinWeiss** [comment]
Better?

**Comment by waehlisch** [comment]
On Fri, 4 Dec 2020, Martine Lenders wrote:

> 
> @miri64 commented on this pull request.
> 
> ___________________________________________________________________________________________________
> 
> In SECURITY.md:
> 
> > +To join the "Trusted RIOT Users" notifications list please send information
> +on the RIOT based service or product to the
> +[security mailing list](security@riot-os.org).
> 
> Why being so vague? IIRC we discussed offline that this "Trusted RIOT Users" notification list will
> be a dedicated category at https://forum.riot-os.org. IMHO we can (and should) be very open about
> this. As such, you should also mention, that we need their forum username with the information they
> send.
> 
  do we really require that senders need to register for the forum to 
send a security vulnerability? i don't think this is a bad idea!

**Comment by waehlisch** [comment]
i removed my previous comment because i read to quickly. forum registration is only required to be informed about vulnerabilities and not to report a bug. ok.

**Comment by MrKevinWeiss** [comment]
Is there anything prevent this from getting merged?

**Comment by miri64** [comment]
> Is there anything prevent this from getting merged?

The forum is not public yet nor is there such a category or user group. The later we can already do before the forum goes public. Ping @aabadie @bergzand?

**Comment by MrKevinWeiss** [comment]
Forum is open!

**Comment by miri64** [comment]
I created a group and category on the forum. It is currently only visible to @MrKevinWeiss, admins, moderators, and me

**Comment by waehlisch** [comment]
> I created a group and category on the forum. It is currently only visible to @MrKevinWeiss, admins, moderators, and me

why not maintainers but moderators?

**Comment by miri64** [comment]
> why not maintainers but moderators?

Because I set it to Staff and the dedicated group for now. But true, it makes sense for maintainers to see and edit that category as well.

**Comment by miri64** [comment]
Done.

**Comment by miri64** [comment]
I can't make the group visible to maintainers out-right though, since the only options are "group members", "group owners", "admins", and "moderators". Unless I add every maintainer to the "trusted users" group by hand of course... or make that group visible to everyone... But I fear that this could generate some false sense of special status within the community and runs a bit against the disclosure part of what we are trying to achieve here.

**Review by miri64 — APPROVED** [review]
But we can iron out those technical details somewhere else or at a later point: back to the document at hand: IMHO this is ready to merge. Here as well: more details can be amended later. ACK!

File: SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+point release, though this may vary depending on the severity and ability of
+trusted RIOT users to provide the fix.
+
+## Trusted RIOT Users
+
+To access the "Trusted RIOT Users" notifications on the
+[RIOT forum](https://forum.riot-os.org) please send information
+on the RIOT based service or product as well as your
+[forum](https://forum.riot-os.org) username to the
+[security mailing list](security@riot-os.org).
```
Comment (waehlisch) [inline SECURITY.md:27]: [security mailing list](mailto:security@riot-os.org)

**Comment by MrKevinWeiss** [comment]
Thanks!

**Pull request merged**